src/main/scala/viper/gobra/backend/Silicon.scala

```diff
@@ -23,18 +22,27 @@ class Silicon(commandLineArguments: Seq[String]) extends ViperVerifier {
       val backend: silicon.Silicon = silicon.Silicon.fromPartialCommandLineArguments(commandLineArguments, reporter)
       
       val startTime = System.currentTimeMillis()
-      backend.start()
-      val result = backend.verify(program)
-      backend.stop()
+      try {
+        backend.start()
+        val result = backend.verify(program)
```

Comment:
We should minimize the amount of code inside the try block.
Why don't we put the try/catch around `backend.verify(program)` instead of putting all the code inside it?
Sth like
```scala
val result = 
    try {
        backend.verify(program)
    } catch ...
```
try/catch blocks are [expressions too](https://docs.scala-lang.org/overviews/scala-book/functional-error-handling.html).

Reply:
The error is thrown at the backend.start() line because it is there where the Z3 process starts.